admin/app/components/solidus_admin/main_nav_component.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module SolidusAdmin
+  # Renders the main navigation of Solidus Admin.
+  class MainNavComponent < BaseComponent
+    def initialize(main_nav_item_component: component("main_nav_item"), items: container.within_namespace("main_nav"))
```

Comment:
If we would add a `components` helper that uses `container.within_namespace` internally, we could make this much more userfriendly and would hide the technical detail of a `Dry::Container` from the user.

```suggestion
    def initialize(main_nav_item_component: component("main_nav_item"), items: components("main_nav"))
```

WDYT?

Reply:
Please, notice that `container.within_namespace("main_nav")` is not resolving `ViewContainer` classes nor instances, but instances of `SolidusAdmin::MainNavItem` entities. I don't think it's a big problem exposing the container (happy to rename it to registry) to the components, as it's precisely intended to resolve dependencies. However, I agree that there's a problem having it also available in the templates, but I can't see a way to avoid that, as it's more of a `view_component` limitation. What do you think?

On top of that, I'd love to use a more declarative specification of dependencies via [`dry-system`'s auto-injection](https://dry-rb.org/gems/dry-system/1.0/dependency-auto-injection/) feature. However, that would require having to call `super` everytime we define the `initialize` method for a component. I didn't go that way because I think it's good to stick as much as we can to bare `view_components` so that ours users don't have a hard time extending our components.

```ruby
module SolidusAdmin
  class MainNavComponent < BaseComponent
    include Import["main_nav_item_component"]

    def initialize(items: container.within_namespace("main_nav"))
      super
      @items = items
    end

    erb_template <<~ERB
      <nav>
        <%=
          render main_nav_item_component.with_collection(
            sorted_items
          )
        %>
      </nav>
    ERB

    private

    def sorted_items
      @items.sort_by(&:position)
    end
  end
end
```

Reply:
Finally, I also added auto-injection to this spike. In the end, `super` needs to be called whenever you override an initializer, and rubocop will complain if you don't.